### ISSUE TITLE
Typescript compiler reports already exported member 'Vec' 

### ISSUE THREAD

**Issue by etjones** [jscad/OpenJSCAD.org#873]
All those Typescript typing files were a lot of work, and I'm grateful to have them. I'm still ending up with a couple issues, though; have you guys seen anything like this? I set up a sample repo for replication. It looks like one issue with a missing file or a reference that should have been removed, and another issue that *might* be resolved with a setting in `tsconfig.json` or something? 

## Expected Behavior
Smooth Typescript compile after  `import jscad from '@jscad/modeling';` in a Typescript file.

## Actual Behavior
Looks like two errors are going on in my install. 

1) Missing file(s) `modeling/src/maths/vec3/unit.*`?  (Reference in `@jscad/modeling/src/maths/vec3/index.d.ts`)
2) The line `type Vec = Vec1 | Vec2 | Vec3` in `operations/transforms/scale.d.ts` (and rotate.d.ts and translate.d.ts) seems to be getting treated as if it were an `export`? I'm not sure why the compiler is complaining about this.

Below is the error output:
``` shell
> npx tsc -p tsconfig.json
The following errors:
node_modules/.pnpm/@jscad+modeling@2.5.0/node_modules/@jscad/modeling/src/maths/vec3/index.d.ts:33:33 - error TS2307: Cannot find module './unit' or its corresponding type declarations.

33 export { default as unit } from './unit'
                                   ~~~~~~~~

node_modules/.pnpm/@jscad+modeling@2.5.0/node_modules/@jscad/modeling/src/operations/transforms/index.d.ts:5:1 - error TS2308: Module './rotate' has already exported a member named 'Vec'. Consider explicitly re-exporting to resolve the ambiguity.

5 export * from './scale'
  ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/.pnpm/@jscad+modeling@2.5.0/node_modules/@jscad/modeling/src/operations/transforms/index.d.ts:7:1 - error TS2308: Module './rotate' has already exported a member named 'Vec'. Consider explicitly re-exporting to resolve the ambiguity.

7 export * from './translate'
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 3 errors.
```

## Steps to Reproduce the Problem
Basically, I'm just trying to to import `@jscad/modeling` in a Typescript file. Stock NPM installs of `@jscad/modeling` and `typescript` and everything.
Minimal demo repo at: https://github.com/etjones/jscad_typescript_export_error

  ``` shell
git clone git@github.com:etjones/jscad_typescript_export_error.git
cd jscad_typescript_export_error
npm install
npx tsc -p tsconfig.json     # <= yields errors
```

## Specifications

  - Version:  @jscad/modeling v2.5.0
  - Platform:  MacOS 11.4 & Windows 10 (June 2021)  (Results identical on each)
  - Environment: (browser, local server, node.js etc)  Node 14.7.1 (MacOS), Node 12.9.0 (Windows)

**Comment by z3dev** [comment]
@danmarshall is there an easy way to verify the typescript files are correct? functional?

**Comment by danmarshall** [comment]
@z3dev - I suppose you could add the TypeScript compiler to the test suite.

**Comment by z3dev** [comment]
The TSD package is included and executed via:

`  "test:tsd": "lerna run test:tsd",`

I guess that’s not the same…
